File: src/objects/File.java
package objects;

public class File {
	public final Program program;
	public final Line[] lines;

	public File(Program program, String content) {
		this.program = program;
		final String[] rawLines = content.split("\n");
		lines = new Line[rawLines.length];
		for(int i = 0; i < lines.length; i++) {
			lines[i] = new Line(this, i + 1, rawLines[i]);
		}
	}
}

File: src/objects/File.kt
package objects

import code.Main
import java.lang.StringBuilder

class File(val project: Project, val name: String, val content: String) {
    val lines: Array<Line>

    init {
        lines = content.split("\n").mapIndexed { i, rawLine -> Line(this, i + 1, rawLine) }.toTypedArray()
    }

    override fun toString(): String {
        val string = StringBuilder()
        for (line in lines)
            string.append("\n").append(line.toString())
        return "File {${Main.indentText(string.toString())}}"
    }
}